tests/Neo.UnitTests/Ledger/UT_MemoryPool.cs

```diff
@@ -241,6 +242,172 @@ public async Task BlockPersistAndReverificationWillAbandonTxAsBalanceTransfered(
             _ = NativeContract.GAS.Mint(applicationEngine, sender, balance, true);
         }
 
+        [TestMethod]
+        public async Task UpdatePoolForBlockPersisted_RemoveBlockConflicts()
+        {
+            // Arrange: prepare mempooled and in-bock txs conflicting with each other.
+            long txFee = 1;
+            var snapshot = GetSnapshot();
+            BigInteger balance = NativeContract.GAS.BalanceOf(snapshot, senderAccount);
+            ApplicationEngine engine = ApplicationEngine.Create(TriggerType.Application, null, snapshot, settings: TestBlockchain.TheNeoSystem.Settings, gas: long.MaxValue);
+            engine.LoadScript(Array.Empty<byte>());
+            await NativeContract.GAS.Burn(engine, UInt160.Zero, balance);
+            _ = NativeContract.GAS.Mint(engine, UInt160.Zero, 6, true); // balance enough for 6 mempooled txs
+
+            var mp1 = CreateTransactionWithFeeAndBalanceVerify(txFee);  // mp1 doesn't conflict with anyone
+            _unit.TryAdd(mp1, engine.Snapshot).Should().Be(VerifyResult.Succeed);
+            var tx1 = CreateTransactionWithFeeAndBalanceVerify(txFee);  // but in-block tx1 conflicts with mempooled mp1 => mp1 should be removed from pool after persist
```

Comment:
@AnnaShaleva,
I see the comment "but in-block tx1 conflicts with mempooled mp1 => mp1 should be removed from pool after persist", but I do not see this test

Reply:
It is below. Ok